### ISSUE TITLE
Ground skills with 0 amotion shouldn't be a thing anymore

### ISSUE THREAD

**Issue by Tokeiburu** [rathena/rathena#7822]
### rAthena Hash

39cdaa6b8

### Client Date

2023-03-30

### Server Mode

Renewal

### Result

While casting ground skills, the character are displaying their entire casting animation and this is caused by the amotion from clif_skill_damage being 0.
```
wd.amotion = (skill_id && skill_get_inf(skill_id)&INF_GROUND_SKILL)?0:sstatus->amotion; //Amotion should be 0 for ground skills.
```
This doesn't seem to be accurate anymore? I wouldn't know when this change would have been made though.

### Expected Result

Ground unit skills should simply use the amotion of the source object.

### How to Reproduce

Use any ground skill.

### Official Information

These values below were extracted from replay files, from various skills. There are more ground skills not covered, those are just the ones I had access to. Some of these were casted from players and others from monsters. So far, three skills appear to have exceptions:
18 - Fire Wall
80 - Fire Pillar
706 - Venom Fog

The file is too large so I'll be posting in pastebin instead (I removed duplicates): https://pastebin.com/t9Ub8UrT

**Comment by Playtester** [comment]
Wouldn't those values depend on your ASPD? You probably should mention what ASPD you measured those values with.

Also what are all those -1 skill_lv packages?

**Comment by Tokeiburu** [comment]
That's correct, the amotion value should depend on your ASPD and that is the issue I'm bringing up. The emulator forcibly sets the amotion to 0 if it's a ground skill when it apparently shouldn't (check the code block in my description above). So my current ASPD value isn't needed for this, since having 200 ASPD is impossible to begin with, so 0 amotion shouldn't be possible either.

This issue is closely related to a ground skill bug where they currently show their full cast animation regardless of your ASPD since it's always forced to 0. And clients, at least newer ones, interpret 0 amotion as "show the full animation", because Gravity logic.

As for those -1, that is the value kRO sets on those packets if the source is a mob. Not all skills display the skill level as -1, it's mostly ground skills if I recall correctly. It's been like this for almost a decade now, who knows why they're doing that.

**Comment by Playtester** [comment]
What I mean is you have proved it's not 0, but to find out what the correct value is, we would need to measure it with different ASPD values. That's why knowing your ASPD on those tests would be useful.
